src/main/java/edu/mit/puzzle/cube/core/model/HuntStatusStore.java

```diff
@@ -327,6 +329,44 @@ public boolean setTeamProperty(
         return updates > 0;
     }
 
+    public <P extends Team.Property> boolean mutateTeamProperty(
```

Comment:
I'm not sure if you're referring to literal Java thread safety, or concurrency / race conditions in general. Let's discuss both.

In terms of literal Java thread safety, I think we're OK here. We're not sharing the connection object we get with anyone else (we still need to address connection pooling - issue #17 - but whatever we do there should be thread-safe). The mutator function is run on the calling thread, as a blocking call, so I don't think there's a threading problem with that either.

For concurrency in general, the problem of incrementing a counter atomically (e.g. incrementing/decrementing a hint token balance) is why I introduced mutateTeamProperty in the first place. Without something like this that does the read+update in a transaction, I think we risk clobbering another request's update and messing up the counter value. So yeah, we should be concerned with concurrency - and that's why this code exists.

That said, I think I got this wrong by leaving it at the default "read committed" isolation level instead of using the "serializable" isolation level (https://www.postgresql.org/docs/current/static/transaction-iso.html). I should also add a retry loop for a commit failure, instead of simply letting it fail. I'll make these changes.